stl/src/vector_algorithms.cpp

```diff
@@ -66,6 +66,14 @@ namespace {
     void _Advance_bytes(const void*& _Target, ptrdiff_t _Offset) noexcept {
         _Target = static_cast<const unsigned char*>(_Target) + _Offset;
     }
+
+    // TRANSITION, DevCom-10331414
+    struct _Zeroupper_on_exit {

```

Comment:
As this is a guard type, we should mark it as `[[nodiscard]]` to prevent classic misuse in the future.